src/runtime/internal/scheduler.ts

```diff
@@ -31,17 +31,21 @@ export function add_flush_callback(fn) {
 	flush_callbacks.push(fn);
 }
 
-let flushing = false;
 const seen_callbacks = new Set();
 export function flush() {
-	if (flushing) return;
-	flushing = true;
+
+	let current_component = null;
+	try {
+		current_component = get_current_component();
+	} catch {
+		// no current component, so leave it as null
```

Comment:
I think we can explain why it's safe to leave it empty, instead of what

Reply:
How often will this throw? Catching errors is expensive.

Reply:
It throws quite often — actually in the majority of cases, I think — so I added a new `maybe_get_current_component` function in https://github.com/sveltejs/svelte/pull/6920/commits/47f90d0a1f30a3e973313bbf8c7544e11ed42993 to avoid throwing and catching useless errors when all we want (in the `flush()` function, at least) is to save and restore the value, without caring what it was.

Reply:
Note that I could also have just had `flush()` close over the `current_component` value from `lifecycle.ts`. It's exported, after all. I wasn't sure if that could produce a stale-closures issue, but I just tried it and it passed all the unit tests, so it's probably never going to be stale. I'll push one more commit getting rid of the `maybe_get_current_component` function.

Reply:
Just pushed https://github.com/sveltejs/svelte/pull/6920/commits/407213cd551f271db30b848b724cbe6430e385b1 which doesn't add a new function, and instead has `flush()` access `current_component` directly rather than calling `get_current_component()` and catching errors.